source/MaterialXContrib/MaterialXMaya/ShadingNodeOverrides.cpp

```diff
@@ -1,22 +1,39 @@
-#include "MaterialXShadingNodeImpl.h"
-
+#include "ShadingNodeOverrides.h"
 #include "MaterialXNode.h"
 #include "Plugin.h"
 #include "MaterialXUtil.h"
+#include "OgsFragment.h"
 
 #include <maya/MDGModifier.h>
 #include <maya/MShaderManager.h>
 #include <maya/MTextureManager.h>
 #include <maya/MPxShadingNodeOverride.h>
 #include <maya/MPxSurfaceShadingNodeOverride.h>
 
-#include <MaterialXData.h>
 #include <MaterialXGenShader/Util.h>
 #include <MaterialXGenShader/HwShaderGenerator.h>
+#include <MaterialXGenOgsXml/OgsXmlGenerator.h>
 #include <MaterialXRender/ImageHandler.h>
 
 namespace mx = MaterialX;
 
+namespace MHWRender
+{
+    class MPxShadingNodeOverride;
+    class MPxSurfaceShadingNodeOverride;
+};
+
+namespace MaterialXMaya
+{
+
+const MString
+    SurfaceOverride::REGISTRANT_ID = "materialXSurface",
```

Comment:
Moved all shading node override classes to this file because SurfaceOverride and TextureOverride were really small.